packages/transitive-overlay/src/index.tsx

```diff
@@ -160,22 +141,25 @@ const TransitiveCanvasOverlay = ({
         type="line"
       />
       <Layer
-        filter={["==", "type", "route"]}
-        id="routes-labels"
-        layout={{
-          "symbol-placement": "line",
-          "text-field": ["get", "name"],
-          "text-keep-upright": true,
-          "text-size": 16
-        }}
+        // Render a solid background of fixed height using the uppercase route name.
+        filter={routeFilter}
+        id="routes-labels-background"
+        layout={getRouteLayerLayout("nameUpper")}
         paint={{
-          "text-color": "#eee",
-          "text-halo-blur": 15,
+          "text-color": ["get", "color"],
           "text-halo-color": ["get", "color"],
-          "text-halo-width": 15
+          "text-halo-width": 4 // Max value is 1/4 of text size per maplibre docs.
         }}
         type="symbol"
       />
+      <Layer
+        // This layer renders transit route names (foreground).
+        filter={routeFilter}
+        id="routes-labels"
+        layout={getRouteLayerLayout("name")}
+        paint={{ "text-color": "#eee" }}
```

Comment:
The text-color here should be imported from the route geojson object

Reply:
Yep, now included with https://github.com/opentripplanner/otp-ui/pull/443/commits/470b95aff6343e6124d13d37698502fe4916ebc2.